lib/vaahextendflutter/services/storage/local/hive/hive_storage_impl.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:hive/hive.dart';
+
+import '../../storage.dart';
+
+/// A class implementing Storage interface using Hive as storage backend.
+class HiveStorageImpl implements Storage {
+  final String name;
+
+  Box? _box;
+
+  HiveStorageImpl({required this.name});
+
+  @override
+  Future<void> init() async {
+    _box = await Hive.openBox(name);
+  }
+
+  @override
+  Future<void> create({required String key, required String value}) async {
+    assert(_box != null, 'Box is null, not initiized.');
+
+    _box!.put(key, value);
+  }
+
+  @override
+  Future<void> createAll({required Map<String, String> values}) async {
+    assert(_box != null, 'Box is null, not initiized.');
+
+    for (String k in values.keys) {
+      await _box!.put(k, values[k]);
+    }
+  }
+
+  @override
+  Future<String?> read({required String key}) async {
+    assert(_box != null, 'Box is null, not initiized.');
+
+    String? result = _box!.get(key);
+    return result;
+  }
+
+  @override
+  Future<Map<String, String?>> readAll({List<String> keys = const []}) async {
```

Comment:
are we gonna support only String as value? As I see force conversion using .toString()

Reply:
Yes, we are, and we suggest that users use a JSON string when the data is complex.
The `.toString()` is used since the hive box is a `Map<dynamic, dynamic>` and we must return a map of type `Map<String, String?>`.